Tooltip.jsx

```diff
@@ -3,178 +3,152 @@
  * @flow strict
  */
 
-/**
- * A wrapper around rc-tooltip that code splits off the third party code into a
- * separate bundle. rc-tooltip will be downloaded only if it is needed to render
- * a tooltip. Use this instead of rc-tooltip.
- */
-
 import * as React from "react";
-import invariant from "./tools/invariant";
-import {
-  makeCancelable,
-  type CancelablePromise,
-} from "./tools/cancelablePromise";
-
-type Props = {
-  +children: React.Node,
-  +placement?: string,
-  +defaultVisible?: boolean,
-  +visible?: boolean,
-  +mouseEnterDelay?: number,
-};
-
-// TODO(dmnd): rc-tooltip doesn't have types :(
-type RcTooltipT = React.ComponentType<Props>;
-
-// Module state. We only want to load RcTooltip once, so this state is stored
-// at the module level.
-let RcTooltip: null | RcTooltipT = null;
-let waitForRcTooltipResolve = null;
-let waitForRcTooltipReject = null;
-
-function initModuleState() {
-  return new Promise((resolve, reject) => {
-    waitForRcTooltipResolve = resolve;
-    waitForRcTooltipReject = reject;
-  });
+import {StyleSheet, css} from "aphrodite";
+import TooltipTrigger from "react-popper-tooltip";
+
+import colors from "./colors";
+import {zIndices} from "./tools/zIndices";
+
+const verticalOffset = 8;
+
+type Props = {|
+  +placement?: "top" | "bottom" | "left" | "right",
+  +overlay: React.Node,
+  +trigger?: "hover" | "click",
+  +mouseEnterDelay?: number, // seconds
+  +mouseExitDelay?: number, // seconds
+  +maxWidth?: number,
+  +children: string | React.Node,
+  +triggerClassName?: string,
+|};
+
+export default function Tooltip({
+  placement = "right",
+  overlay,
+  trigger = "hover",
+  children,
+  mouseEnterDelay = 0,
+  mouseExitDelay = 0,
+  maxWidth = null,
+  triggerClassName = "",
+}: Props) {
+  return (
+    <TooltipTrigger
+      placement={placement}
+      trigger={trigger}
+      tooltip={MakeContainer(overlay, maxWidth)}
+      delayShow={mouseEnterDelay * 1000}
+      delayHide={mouseExitDelay * 1000}
+      modifiers={{
+        offset: {
+          offset: `0,${verticalOffset}px`,
+        },
+      }}
+    >
+      {MakeTrigger(children, triggerClassName)}
+    </TooltipTrigger>
+  );
 }
-let waitForRcTooltip = initModuleState();
 
-/**
- * Reset module state. Only useful during testing.
- */
-export function insideTestResetModuleState() {
-  RcTooltip = null;
-  waitForRcTooltipResolve = null;
-  waitForRcTooltipReject = null;
-  waitForRcTooltip = initModuleState();
+function MakeContainer(children: React.Node, maxWidth: ?number) {
```

Comment:
Also not a React component

```suggestion
function makeContainer(children: React.Node, maxWidth: ?number) {
```

Reply:
Fixed